clash-prelude/src/Clash/Signal/Delayed/Bundle.hs

```diff
@@ -219,3 +219,20 @@ instance KnownNat d => Bundle (RTree d a) where
   type Unbundled t delay (RTree d a) = RTree d (DSignal t delay a)
   bundle   = sequenceA
   unbundle = sequenceA . fmap lazyT
+
+-- | Internal data type, see "TaggedUnit".
+data TaggedUnit' (d :: k) (dom :: i) = TaggedUnit
+
+-- | Same as "Clash.Signal.Bundle", but adapted for "DSignal".
+type TaggedUnit (d :: Nat) (dom :: Domain) = TaggedUnit' '(d, d) '(dom, dom)
+
+-- | See https://github.com/clash-lang/clash-compiler/pull/539/commits/94b0bff5770aa4961e04ddce2515130df3fc7863
+-- and documentation for "TaggedUnit".
+instance Bundle (TaggedUnit' d1 dom1) where
```

Comment:
Are you allowed to do:

```haskell
instance Bundle (TaggedUnit' '(d1,dom1)) where
```

And use only one `TaggedUnit'`?

Reply:
I'm quite sure that's possible. Is that a "changes requested"?

Reply:
Yes, since we only need one `TaggedUnit'` then instead of two

Reply:
And if we remove the `TaggedUnit'/TaggedUnit` distinction, we’d have only a single `TaggedUnit` as well. Which I wouldn’t mind given that its use is quite niche: i.e. I wonder how much it changes user-code outside of type class instances

Reply:
I don't want users writing / reading `TaggedUnit' '(dom1,dom1)`, so I'd like to keep `TaggedUnit` around. Merging the `TaggedUnit'`s is fine with me.

Reply:
Tot add to that: it does turn up in code outside of type class instances.